build.gradle

```
@@ -107,24 +127,44 @@ List<String> resolveIntelliJVersions(String versionKey) {
   return [versionValue]
 }
 
-class ResolveIntelliJVersions extends DefaultTask {
-  @Input
-  @Option(option = 'version-key', description = 'Either release number (like 2020.3) or key of intellijVersions (like eapOfLatestSupportedMajor)')
-  String versionKey
-
-  @OptionValues('version-key')
-  List<String> getAvailableVersionKeys() {
-    return new ArrayList<>(project.intellijVersions.keySet())
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.Document
+
+String checkForNewerEap(String intellijMajorVersion) {
```

I'd specify the name, IIUC it should be sth like:
```suggestion
String checkForEapForMajorVersionHigherThan(String intellijMajorVersion) {
```

---

yes this method name is also better for me, but I don't know exactly what it would look like with the word "then" in a situation where eapOfLatestSupportedMajor is null in properties file -> in this situation I also pass null as method param so we have checkForEapForMajorVersionHigherThan (null) ..?

---

I'd do it so (pseudocode):

```python
if latestEap != null:
  newerEap = project.checkForEapForMajorVersionHigherThan(latestEap) // we're checking the *full* EAP version! so comparing e.g `222.2137.1 > 222.1920.123`
else:
  newerEap = project.checkForEapForMajorVersionHigherThan(intellijVersionToBuildNumber(latestStable)) // here we'd compare e.g. `222.2137.1 > 221`, smart comparison algorithm should be devised for such version comparison - maybe there's something ready in Gradle?